hail/python/hailtop/batch/backend.py

```diff
@@ -51,6 +51,24 @@ class Backend(abc.ABC, Generic[RunningBatchType]):
 
     _closed = False
 
+    def __init__(self):
+        self._batch_fses: Dict["batch.Batch", "RouterAsyncFS"] = {}
+
+    def batch_fs(self, batch: "batch.Batch") -> "RouterAsyncFS":
+        try:
+            return self._batch_fses[batch]
+        except KeyError:
+            if batch.requester_pays_project is not None:
+                self._batch_fses[batch] = RouterAsyncFS(
+                    gcs_kwargs={"gcs_requester_pays_configuration": batch.requester_pays_project}
+                )
+                return self._batch_fses[batch]
+            return self._fs
+
+    @abc.abstractmethod
+    def validate_file(self, uri: str, batch: Optional["batch.Batch"] = None) -> None:
```

Comment:
It is more intuitive to me that a `validate_file` method would take some configuration related to GCS rather than a whole `Batch` object.

Reply:
updated